abstract-bits-derive/src/model.rs

```diff
@@ -187,18 +196,61 @@ impl Field {
     }
 }
 
-fn max_size_from_control_list(ident: &syn::Ident, previous_fields: &[Field]) -> usize {
-    if let Some(bits) = previous_fields.iter().find_map(|f| match f {
-        Field::ControlList {
-            controlled, bits, ..
-        } if controlled == ident => Some(bits),
+fn max_size_from_controller_field(
+    controller_expr: &syn::Expr,
+    previous_fields: &[Field],
+) -> usize {
+    // Extract the base field name from the expression
+    let controller_ident = match controller_expr {
+        syn::Expr::Path(path) if path.path.segments.len() == 1 => {
+            &path.path.segments[0].ident
+        }
+        syn::Expr::Field(field_expr) => {
+            if let syn::Expr::Path(base_path) = &*field_expr.base {
+                if base_path.path.segments.len() != 1 {
+                    abort!(
+                        controller_expr.span(),
+                        "Complex controller expressions not yet supported"
+                    );
+                }
+
+                &base_path.path.segments[0].ident
+            } else {
+                abort!(
+                    controller_expr.span(),
+                    "Complex controller expressions not yet supported"
+                );
+            }
+        }
+        _ => abort!(
+            controller_expr.span(),
+            "Controller expression must be a field name or field access"
+        ),
+    };
+    
+    // Look for the controller field in previous_fields
```

Comment:
maybe extract this to `controller_in_previous_fields(previous_fields)`. Could return an Option and abort here or just abort in the function.